sale_configurator_option/tests/test_sale_order.py

```diff
@@ -40,3 +52,23 @@ def test_conf_total_amount_main_without_price(self):
         self.assertEqual(self.line_opt_1.price_config_total, 0)
         self.assertEqual(self.line_opt_2.price_config_total, 0)
         self.assertEqual(self.line_opt_3.price_config_total, 0)
+
+    def test_conf_product_change_option(self):
+        self.line_with_opt_2.product_id = self.product_with_option_2
+        self.line_with_opt_2.product_id_change()
+        default_options = [
+            x.product_id
+            for x in self.product_with_option_2.product_tmpl_id.\
+                    product_config_opt_ids
+            if x.opt_default_qty > 0]
+        self.assertEqual(len(self.line_with_opt_2.option_ids), len(default_options))
```

Comment:
Something like that will be better
product_ids = set(line.mapped("option_ids.product_id.id"))
self.assertEqual(product_ids, set([self.product_option_1.id, self.product_option_2.id])

Reply:
self.assertEqual(len(line.option_ids), 2)